src/electionguard_gui/components/create_election_component.py

```diff
@@ -0,0 +1,117 @@
+from os import path
+from shutil import make_archive, rmtree
+import tempfile
+from typing import Any
+import eel
+from electionguard.constants import get_constants
+from electionguard_cli.setup_election.output_setup_files_step import (
+    OutputSetupFilesStep,
+)
+from electionguard_cli.setup_election.setup_election_builder_step import (
+    SetupElectionBuilderStep,
+)
+from electionguard_gui.eel_utils import eel_fail, eel_success
+from electionguard_gui.components.component_base import ComponentBase
+from electionguard_gui.services import (
+    KeyCeremonyService,
+    GuiSetupInputRetrievalStep,
+    ElectionService,
+)
+
+
+class CreateElectionComponent(ComponentBase):
+    """Responsible for functionality related to creating encryption packages for elections"""
+
+    _COMPRESSION_FORMAT = "zip"
+
+    _key_ceremony_service: KeyCeremonyService
+    _setup_input_retrieval_step: GuiSetupInputRetrievalStep
+    _setup_election_builder_step: SetupElectionBuilderStep
+    _election_service: ElectionService
+    _output_setup_files_step: OutputSetupFilesStep
+
+    def __init__(
+        self,
+        key_ceremony_service: KeyCeremonyService,
+        election_service: ElectionService,
+        setup_input_retrieval_step: GuiSetupInputRetrievalStep,
+        setup_election_builder_step: SetupElectionBuilderStep,
+        output_setup_files_step: OutputSetupFilesStep,
+    ) -> None:
+        self._key_ceremony_service = key_ceremony_service
+        self._setup_input_retrieval_step = setup_input_retrieval_step
+        self._setup_election_builder_step = setup_election_builder_step
+        self._election_service = election_service
+        self._output_setup_files_step = output_setup_files_step
+
+    def expose(self) -> None:
+        eel.expose(self.get_keys)
+        eel.expose(self.create_election)
+
+    def get_keys(self) -> dict[str, Any]:
+        self._log.debug("Getting keys")
+        db = self._db_service.get_db()
+        completed_key_ceremonies = self._key_ceremony_service.get_completed(db)
+        keys = [
+            key_ceremony.to_id_name_dict() for key_ceremony in completed_key_ceremonies
+        ]
+        return eel_success(keys)
+
+    def create_election(
```

Comment:
This method is too long and complicated.  I'll look for options to refactor when I next touch it.